### DISCUSSION TITLE
Q-select not returning the right value

### DISCUSSION THREAD

**Discussion by lamseck-jonathan** [quasarframework/quasar#15694] [General - Components / Directives / etc]
I am using a q-select and the option-value is not working well, it returns the whole object instead of the property that i ask him to 
<q-select 
        class="col-3 q-pl-sm q-pr-sm"
        label="Portefeuille"
        hide-bottom-space
        outlined 
        dense 
        v-model="transactionItem.portefeuille_id" 
        :options="portefeuilleList" 
        option-value="id"
        :option-label="(item:Portefeuille) => item.nom_portefeuille"
        />
        
        here is the q-select
        
        <q-select 
        class="col-3 q-pl-sm q-pr-sm"
        label="Portefeuille"
        hide-bottom-space
        outlined 
        dense 
        v-model="transactionItem.portefeuille_id" 
        :options="portefeuilleList" 
        :option-value="(item:Portefeuille) => item.id"
        :option-label="(item:Portefeuille) => item.nom_portefeuille"
        />
        
        i tryed the two above but nothing worked , i am using quasar 2 , vite , vue , and typescript

**Comment by dongwa** [comment]
You can try `emit-value` prop :

![image](https://user-images.githubusercontent.com/38457491/231039880-41b51990-dcf0-4e07-ae4a-0cc08cabeeb7.png)

And then,you may be wondering, why it show value isn't lable?
You can use `map-options` prop to solve this:

![image](https://user-images.githubusercontent.com/38457491/231040976-8ac8c64e-e088-4eed-b4f8-6f330b530f0e.png)
